src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -906,6 +906,31 @@ public void GetAsync_ServerNeedsAuthAndNoCredential_StatusCodeUnauthorized()
             }
         }
 
+        [Fact]
+        public async Task HttpClientHandler_CredentialIsNotCredentialCacheAfterRedirect_StatusCodeOK()
+        {
+            HttpClientHandler handler = CreateHttpClientHandler();
+            handler.Credentials = _credential;
+            using (var client = new HttpClient(handler))
+            {
+                Uri redirectUri = Configuration.Http.RedirectUriForCreds(
+                    secure: false,
+                    statusCode: 302,
+                    userName: Username,
+                    password: Password);
+                using (HttpResponseMessage unAuthResponse = await client.GetAsync(redirectUri))
+                {
+                    Assert.Equal(HttpStatusCode.Unauthorized, unAuthResponse.StatusCode);
+                }
+
+                Uri uri = Configuration.Http.BasicAuthUriForCreds(secure: true, userName: Username, password: Password);
```

Comment:
Why have these two test cases like this using duplicated code except for the "secure" variable?  it would be better as a Theory with InlineData for the "secure" part.

Reply:
These are not 2 separate tests. The test is to verify after redirection the handler credentials are not cleared, which was the bug.